src/components/attach/MediaHub.js

```diff
@@ -0,0 +1,92 @@
+import MessageAttachment from "@components/message/elements/MessageAttachment";
+import getFileType from "@utils/media/get_file_type";
+import removeAndNavigateLastSection from "@utils/navigation/get_prev_page";
+import { selectMessagesEntities } from "@store/values/Messages";
+import { useLocation } from "react-router-dom";
+import { useMemo, useState } from "react";
+import { useSelector } from "react-redux";
+
+import "@styles/attach/MediaHub.css";
+
+import { ReactComponent as Prev } from "@icons/options/Prev.svg";
+import { ReactComponent as Next } from "@icons/options/Next.svg";
+
+export default function MediaHub() {
+  const { pathname, hash } = useLocation();
+
+  const [currentIndex, setCurrentIndex] = useState(() => {
+    const [, , index] = hash.split("=");
+    return +index;
+  });
+
+  const mid = hash.split("=")[1];
+  const messages = useSelector(selectMessagesEntities);
+  const attachments = useMemo(() => {
+    const currentMessage = messages[mid];
+    return currentMessage ? currentMessage.attachments : [];
+  }, [messages, mid]);
```

Comment:
instead of getting messages array from redux and then access that particular message via mid, can we just get that single message from store via selector?